crates/driver/src/util/math.rs

```diff
@@ -0,0 +1,18 @@
+use ethereum_types::U256;
+
+/// Computes `x * q / d` rounding down.
+///
+/// Returns `None` if `d` is `0` or if the result overflows a 256-bit integer.
+pub fn mul_ratio(x: U256, q: U256, d: U256) -> Option<U256> {
+    x.full_mul(q).checked_div(d.into())?.try_into().ok()
+}
+
+/// Computes `x * q / d` rounding up.
+///
+/// Returns `None` if `d` is `0` or if the result overflows a 256-bit integer.
+pub fn mul_ratio_ceil(x: U256, q: U256, d: U256) -> Option<U256> {
+    let p = x.full_mul(q);
+    let result = U256::try_from(p.checked_div(d.into())?).ok()?;
+    let round_up = !p.checked_rem(d.into())?.is_zero();
+    result.checked_add((round_up as u8).into())
```

Comment:
nit: there is `u8::from(bool)` to avoid the more unsafe `as` cast. Doesn't really make a difference in this case but IMO this is a best practice.

Reply:
https://github.com/cowprotocol/services/pull/1902/commits/43aede4515bf55c990d0d7f2d7180da82537fb0e